server/app/services/program/ActiveAndDraftPrograms.java

```diff
@@ -27,27 +27,46 @@ public final class ActiveAndDraftPrograms {
 
   /**
    * Queries the existing active and draft versions and builds a snapshotted view of the program
-   * state.
+   * state. Since a ProgramService argument is included, we will get the full program definition,
+   * which includes the question definitions.
    */
   public static ActiveAndDraftPrograms buildFromCurrentVersions(
       ProgramService service, VersionRepository repository) {
     return new ActiveAndDraftPrograms(
-        service, repository.getActiveVersion(), repository.getDraftVersion());
+        repository.getActiveVersion(), repository.getDraftVersion(), Optional.of(service));
```

Comment:
**suggestion (blocking):** for both public static methods, rename them so the method name is explicit around the behavior with the `ProgramService` and syncing questions. That way callers will know what's happening without having to read the comments.

Reply:
Updated to `buildFromCurrentVersionsWithoutQuestionLoad` and `buildFromCurrentVersionsWithProgramServiceQuestionLoad`. Those names are pretty long though so open to other name ideas. I was thinking buildFromCurrentVersions Shallow and Deep could work as well. Let me know what you think.

Reply:
synced / unsyced? that would match the verb used inside program service `syncProgram` etc

Reply:
Sounds good, updated!